include/deal.II/distributed/cell_data_transfer.h

```diff
@@ -0,0 +1,346 @@
+      /**
+       * Whenever data shall be transferred across coarsening, it is not a
+       * trivial choice how to process data of cells which will be coarsened. Or
+       * in other words, which data should be stored in the corresponding parent
+       * cell.
+       *
+       * In this namespace, we offer a few strategies that cope with this
+       * problem, that can be passed to the
+       * parallel::distributed::CellDataTransfer constructor.
+       */
+      struct CoarseningStrategies
+      {
+        /**
+         * Evaluate data from @p input_vector on children of @p parent.
+         * Check if data on all children match, and return that value.
+         */
+        static typename VectorType::value_type
+        check_equality(const typename parallel::distributed::
+                         Triangulation<dim, spacedim>::cell_iterator &parent,
+                       const VectorType *input_vector)
+        {
+          const typename VectorType::value_type value =
+            (*input_vector)[parent->child(0)->active_cell_index()];
+
+          for (unsigned int child_index = 1; child_index < parent->n_children();
+               ++child_index)
+            Assert(value == (*input_vector)[parent->child(child_index)
+                                              ->active_cell_index()],
+                   ExcMessage(
+                     "Values on cells that will be coarsened are not equal!"));
+
+          return value;
+        }
+
+        /**
+         * Evaluate data from @p input_vector on children of @p parent.
+         * Return sum.
+         */
+        static typename VectorType::value_type
+        sum(const typename parallel::distributed::Triangulation<dim, spacedim>::
+              cell_iterator & parent,
+            const VectorType *input_vector)
+        {
+          typename VectorType::value_type sum = 0;
+
+          for (unsigned int child_index = 0; child_index < parent->n_children();
+               ++child_index)
+            sum +=
+              (*input_vector)[parent->child(child_index)->active_cell_index()];
+
+          return sum;
+        }
+
+        /**
+         * Evaluate data from @p input_vector on children of @p parent.
+         * Return mean value.
+         */
+        static typename VectorType::value_type
+        mean(const typename parallel::distributed::
+               Triangulation<dim, spacedim>::cell_iterator &parent,
+             const VectorType *                             input_vector)
+        {
+          return sum(parent, input_vector) / parent->n_children();
+        }
+      };
+
+      /**
+       * Constructor.
+       *
+       * @param[in] triangulation The triangulation on which all operations will
+       *   happen. At the time when this constructor is called, the refinement
+       *   in question has not happened yet.
+       * @param[in] transfer_variable_size_data Specify whether size of data to
+       *   transfer varies from cell to cell.
```

Comment:
can you elaborate here what that means? I mean, I'm passing a `std::vector<unsigned int>` in, that looks pretty constant to me... :-)